DIPs/DIP1038.md

```diff
@@ -242,148 +202,99 @@ possible balance among the following goals:
    makes future improvements to both the language and its compilers more
    difficult to achieve.
 
-3. **Lack of friction.** A language feature is only useful in practice to the
-   extent that programmers choose to use it. If a feature places any
-   obstacles in the programmer's way (such as additional compile-time errors),
-   it must offer benefits commensurate with that cost.
+3. **Rigor.** A language feature that provides strong guarantees that
+   programmers can rely on is more useful that one that provides weak
+   guarantees or permits exceptions and special cases. Rigorous language
+   features compose more easily than non-rigorous ones.
 
 A few possible alternative designs, along with the reasons for their rejection,
 are discussed below.
 
-#### Dataflow analysis
-
-As noted above, the design presented in this DIP allows "false negatives":
-expressions in which the return value of a `@nodiscard`-annotated function is
-discarded without triggering a compile-time error.
-
-One way to avoid false negatives would be to have the compiler perform dataflow
-analysis to determine which expressions could potentially evaluate to the
-return value of a `@nodiscard`-annotated function, and issue an error if any of
-those expressions were discarded.
+#### `@mustUse` as a function attribute
 
-Whole-program dataflow analysis would completely eliminate false negatives, but
-its costs in terms of both implementation complexity and compile-time
-performance would likely be prohibitive. Intra-procedural dataflow analysis, on
-the other hand, would be more feasible to implement, but would still allow
-false negatives in some cases. For example:
+As described above, `@mustUse` is restricted to use with user-defined `struct`,
+`union`, and `enum` types, and cannot easily be applied to functions that
+return other kinds of types. One way to address this shortcoming would be to
```

Comment:
This could discuss the use of anonymous `enum`s as a workaround, e.g.

```d
template MustUse(T)
{
	@mustUse enum MustUse : T;
}

extern(C) MustUse!int someFunction();
```

Reply:
``alias MustUse(T) = @mustUse T;``

Another form, which I commented that I hope works.

Reply:
also that won't work for `extern(C++)` functions

Reply:
> `alias MustUse(T) = @mustUse T;`
> 
> Another form, which I commented that I hope works.

`mustUse` must be attached to the declaration IIUC, so probably not.

> also that won't work for extern(C++) functions

True, those could be handled via inline wrapper functions. But that's beside the point of that example.